tests/foreman/ui/test_ldap_auth.py

```diff
@@ -140,10 +175,63 @@ def test_positive_ad_basic_roles(self):
         :expectedresults: Log in to foreman UI successfully and can access
             appropriate functional areas in UI
 
-        :caseautomation: notautomated
-
         :CaseLevel: System
         """
+        server_name = gen_string('alpha')
+        with Session(self) as session:
+            make_ldapauth(
+                session,
+                name=server_name,
+                server=self.ldap_hostname,
+                server_type=LDAP_SERVER_TYPE['UI']['ad'],
+                login_name=LDAP_ATTR['login_ad'],
+                first_name=LDAP_ATTR['firstname'],
+                surname=LDAP_ATTR['surname'],
+                mail=LDAP_ATTR['mail'],
+                account_user=self.ldap_user_name,
+                account_passwd=self.ldap_user_passwd,
+                account_basedn=self.base_dn,
+                account_grpbasedn=self.group_base_dn,
+                orgs=[entities.Organization().create().name],
+                org_select=True,
+                locations=[entities.Location().create().name],
+                loc_select=True,
+            )
+        with Session(self, user=self.ldap_user_name,
+                     password=self.ldap_user_passwd):
+            pass
+        with Session(self) as session:
+            strategy, value = common_locators['entity_deselect']
+            name = self.ldap_user_name
+            resource_type = 'Architecture'
+            permissions = ['view_architectures']
+            role_name = gen_string('alphanumeric')
+            make_role(session, name=role_name)
+            self.assertIsNotNone(self.role.search(role_name))
+            self.role.add_permission(
+                role_name,
+                resource_type=resource_type,
+                permission_list=permissions,
+            )
+            self.assertIsNotNone(
+                self.role.wait_until_element(
+                    common_locators['alert.success']))
+            assigned_permissions = self.role.get_permissions(
+                role_name, [resource_type])
+            self.assertIsNotNone(assigned_permissions)
+            self.assertEqual(
+                set(permissions), set(assigned_permissions[resource_type]))
+            self.user.update(name, new_roles=[role_name])
+            self.user.click(self.user.search(name))
+            self.user.click(tab_locators['users.tab_roles'])
+            self.assertIsNotNone(
+                self.user.wait_until_element((strategy, value % role_name)))
+        with Session(self, user=self.ldap_user_name,
+                     password=self.ldap_user_passwd):
+            arch_name = gen_string('alphanumeric')
+            entities.Architecture(name=arch_name).create()
```

Comment:
That is invalid statement as here you are going to use API to create entity and it will use admin user to perform operation. Just for your information, correct flow will be:
```
        cfg = get_nailgun_config()
        cfg.auth = (user.login, password)
        ak = entities.ActivationKey(cfg).create()
```

Anyway, use WebUI method here instead API